Sources/VaporJWT/Payload.swift

```diff
@@ -0,0 +1,8 @@
+import Node
+import Auth
+
+/// Types conforming to this protocol
+/// can be stored as payloads in JWTs
+public protocol Payload: NodeConvertible {
+    func makeCredentials() throws -> Auth.Credentials
```

Comment:
Sure. Claims are an official part of the [JWT spec](https://tools.ietf.org/html/rfc7519#section-4) that can further restrict the validity of a token. There is a set of official claims but custom ones can be made as well. I made implementations of the official ones. Claims are stored inside the payload.

An example could be that a token is only valid for 1 hour. To enable that you can use `ExpirationTimeClaim` and specify a time (in seconds) after which the token should be rejected (regardless of signer validity). Other use cases would be to restrict access to a certain audience ('AudienceClaim`).

As an example let's enforce that a token is not expired and was issued by "vapor". To enforce this you'd do: `let valid = try jwt.verifyClaims([ExpirationTimeClaim(), IssuerClaim("vapor")])`. By default, the ExpirationTimeClaim uses the current date to compare against. This reason this works is that JWTs cannot be tampered with without invalidating the signature.

In light of your changes to signature verification, claim validation will need a bit of an update too.